mm2src/lp_swap/taker_swap.rs

```diff
@@ -673,8 +673,9 @@ impl TakerSwap {
     }
 
     async fn start(&self) -> Result<(Option<TakerSwapCommand>, Vec<TakerSwapEvent>), String> {
+        // do not use self.r().data here as it is not initialized at this step yet
         let stage = FeeApproxStage::StartSwap;
-        let dex_fee = dex_fee_amount_from_taker_coin(&self.taker_coin, &self.r().data.maker_coin, &self.taker_amount);
+        let dex_fee = dex_fee_amount_from_taker_coin(&self.taker_coin, self.maker_coin.ticker(), &self.taker_amount);
```

Comment:
I see this error appeared due to my changes 😅

Reply:
It's NP, I think we need to refactor it one day to prevent using uninitialized data at compile-time if possible.